internal/cmd/restorer.go

```diff
@@ -0,0 +1,350 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
+	"github.com/cenkalti/backoff/v4"
+	"github.com/mattn/go-isatty"
+	"github.com/rs/zerolog/log"
+	"github.com/samber/lo"
+	"github.com/schollz/progressbar/v3"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/authzed/zed/internal/client"
+	"github.com/authzed/zed/pkg/backupformat"
+)
+
+// Fallback for CRDB datastore on SpiceDB < 1.29.0 not returning proper gRPC codes
```

Comment:
what about the other datastore kinds?

Reply:
I didn't test all of them. I'll look into it.

Reply:
I tested all datastores and added codes to handle when it was missing